src/main/pages/setup-docker/docker-config.adoc

```diff
@@ -1,56 +1,57 @@
+By default, Che lets Docker publish exposed ports in a random manner. Docker chooses available ports from the ephemeral port range to expose workspace link:servers.html[servers]. This requires the opening of the ephemeral port range (and the Keycloak `5050` port for multi-user Che) to the world.
+
+To run Che in single-port mode, add `-e CHE_SINGLE_PORT=true` to the run syntax. In this case, a Traefik container will be used to route traffic through a single port.
+
+*Using Wildcard DNS*
 
-To run Che in a single port mode add `-e CHE_SINGLE_PORT=true` to your run syntax. In this case, a Traefik container will be used to route traffic through a single port.
+In single-port mode, Che builds URLs of workspace services using the following pattern: `serviceName-machineName-ws-ID.IP.wildcardDNSProvider`. For example, with an external IP of `193.12.34.56`, the URL of a workspace agent will be `wsagent-http-dev-machine-workspace0bcgkgkvsqi31b4u.193.12.34.56.nip.io`.
 
-*Wildcard DNS*
+* By default, http://nip.io/[nip.io] is used. This is an external wildcard DNS provider. The nip.io service must be available for networking in single-port Che to work correctly.
 
-In a single port mode, Che builds URLs of workspace services using the following pattern - `serviceName-machineName-ws-ID.IP.wildcardDNSProvider`. So, if your external IP is `193.12.34.56`, URL of a workspace agent will look like `wsagent-http-dev-machine-workspace0bcgkgkvsqi31b4u.193.12.34.56.nip.io`
+* You can use a different wildcard DNS provider with the `CHE_SINGLEPORT_WILDCARD__DOMAIN_HOST` environment variable.
 
-* by default http://nip.io/[nip.io] is used. This is an external wildcard DNS provider, and if it is down for some reason, networking in single port Che is broken.
-* you can use a different wildcard DNS provider with `pass:[CHE_SINGLEPORT_WILDCARD__DOMAIN_HOST]` env.
-* if you don’t want the external IP to be part of the url (`serviceName-machineName-ws-ID.wildcardDNSProvider`, for example to use a wildcard SSL certificate), you can specify `pass:[CHE_SINGLEPORT_WILDCARD__DOMAIN_IPLESS=true]` with a custom wildcard DNS (e.g. `pass:[CHE_SINGLEPORT_WILDCARD__DOMAIN_HOST=domain.tld]`. Be aware that you need to have a matching DNS entry for `*.domain.tld`. If you are using the multi-user mode, Keycloak will be provided at `keycloak.domain.tld`.
+* To exclude the external IP from the URL (`serviceName-machineName-ws-ID.wildcardDNSProvider`, for example to use a wildcard SSL certificate), specify `CHE_SINGLEPORT_WILDCARD_DOMAIN_IPLESS=true` with a custom wildcard DNS (example: `CHE_SINGLEPORT_WILDCARD_DOMAIN_HOST=domain.tld`). You must have a matching DNS entry for `*.domain.tld`. If you are using the multi-user mode, Keycloak will be provided at `keycloak.domain.tld`.
```

Comment:
Considering the warning in the original sentence how about adding the ' you must have...' as a note?

Reply:
@Preeticp , I think we considered that and thought it wasn't required. This was when @rkratky  and Robin reviewed this. Leaving it as is for now.